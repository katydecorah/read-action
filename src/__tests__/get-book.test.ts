import getBook from "../get-book";
import { exportVariable } from "@actions/core";
import { promises, readFileSync } from "fs";
import book from "./fixture.json";
import isbn from "node-isbn";
import * as core from "@actions/core";

const books = readFileSync("./_data/read.json", "utf-8");
const dateFinished = "2020-09-12";

jest.mock("@actions/core");

const defaultOptions = {
  readFileName: "_data/read.yml",
  requiredMetadata: "title,pageCount,authors,description",
  timeZone: "America/New_York",
};

describe("getBook", () => {
  beforeEach(() => {
    jest
      .spyOn(core, "getInput")
      .mockImplementation((v) => defaultOptions[v] || undefined);
  });

  test("works", async () => {
    jest.spyOn(isbn, "resolve").mockResolvedValueOnce(book);
    jest.spyOn(promises, "readFile").mockResolvedValueOnce(books);
    await getBook({
      dates: {
        dateAdded: undefined,
        dateStarted: undefined,
        dateFinished,
      },
      bookIsbn: "9780525658184",
      providers: ["google"],
      bookStatus: "finished",
      fileName: "_data/read.yml",
    });
    expect(exportVariable.mock.calls).toMatchInlineSnapshot(`
      [
        [
          "BookTitle",
          "Transcendent Kingdom",
        ],
        [
          "BookThumbOutput",
          "book-9780525658184.png",
        ],
        [
          "BookThumb",
          "https://books.google.com/books/content?id=ty19yQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
        ],
      ]
    `);
    jest.spyOn(promises, "readFile").mockResolvedValueOnce(books);
    expect(
      await getBook({
        dates: {
          dateAdded: undefined,
          dateStarted: undefined,
          dateFinished,
        },
        bookIsbn: "9780525658184",
        providers: ["google"],
        bookStatus: "finished",
        fileName: "_data/read.yml",
      })
    ).toMatchSnapshot();
  });
  test("fails", async () => {
    jest
      .spyOn(isbn, "resolve")
      .mockRejectedValue(new Error("Request failed with status code 401"));
    await expect(
      getBook({
        dates: {
          dateAdded: undefined,
          dateStarted: undefined,
          dateFinished,
        },
        bookIsbn: "9780525658184",
        providers: ["google"],
        bookStatus: "finished",
        fileName: "_data/read.json",
      })
    ).rejects.toMatchInlineSnapshot(
      `[Error: Book (9780525658184) not found. Request failed with status code 401]`
    );
  });
});
