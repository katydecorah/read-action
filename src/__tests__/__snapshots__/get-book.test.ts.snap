// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBook works 2`] = `
[
  {
    "authors": [
      "Raven Leilani",
    ],
    "dateFinished": "2020-09-06",
    "description": "Sharp, comic, disruptive, tender, Raven Leilani's debut novel, Luster, sees a young black woman fall into art and someone else's open marriage. Edie is stumbling her way through her twenties--sharing a subpar apartment in Bushwick, clocking in and out of her admin job, making a series of inappropriate sexual choices. She's also, secretly, haltingly, figuring her way into life as an artist. And then she meets Eric, a digital archivist with a family in New Jersey, including an autopsist wife who has agreed to an open marriage--with rules. As if navigating the constantly shifting landscapes of contemporary sexual manners and racial politics weren't hard enough, Edie finds herself unemployed and falling into Eric's family life, his home. She becomes a hesitant friend to his wife and a de facto role model to his adopted daughter. Edie is the only black woman who young Akila knows. Razor sharp, darkly comic, sexually charged, socially disruptive, Luster is a portrait of a young woman trying to make her sense of her life in a tumultuous era. It is also a haunting, aching description of how hard it is to believe in your own talent and the unexpected influences that bring us into ourselves along the way.",
    "isbn": "9780385696005",
    "language": "en",
    "link": "https://books.google.com/books/about/Luster.html?hl=&id=eJ06zQEACAAJ",
    "pageCount": 240,
    "printType": "BOOK",
    "publishedDate": "2020-08-04",
    "thumbnail": "https://books.google.com/books/content?id=eJ06zQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
    "title": "Luster",
  },
  {
    "authors": [
      "Yaa Gyasi",
    ],
    "categories": [
      "Fiction",
    ],
    "dateAdded": undefined,
    "dateFinished": "2020-09-12",
    "dateStarted": undefined,
    "description": "NEW YORK TIMES BEST SELLER • A TODAY SHOW #ReadWithJenna BOOK CLUB PICK! • Finalist for the WOMEN'S PRIZE Yaa Gyasi's stunning follow-up to her acclaimed national best seller Homegoing is a powerful, raw, intimate, deeply layered novel about a Ghanaian family in Alabama. Gifty is a sixth-year PhD candidate in neuroscience at the Stanford University School of Medicine studying reward-seeking behavior in mice and the neural circuits of depression and addiction. Her brother, Nana, was a gifted high school athlete who died of a heroin overdose after an ankle injury left him hooked on OxyContin. Her suicidal mother is living in her bed. Gifty is determined to discover the scientific basis for the suffering she sees all around her. But even as she turns to the hard sciences to unlock the mystery of her family's loss, she finds herself hungering for her childhood faith and grappling with the evangelical church in which she was raised, whose promise of salvation remains as tantalizing as it is elusive. Transcendent Kingdom is a deeply moving portrait of a family of Ghanaian immigrants ravaged by depression and addiction and grief—a novel about faith, science, religion, love. Exquisitely written, emotionally searing, this is an exceptionally powerful follow-up to Gyasi's phenomenal debut.",
    "isbn": "9780525658184",
    "language": "en",
    "link": "https://books.google.com/books/about/Transcendent_Kingdom.html?hl=&id=MvaNEAAAQBAJ",
    "printType": "BOOK",
    "publishedDate": "2020-09-01",
    "status": "finished",
    "thumbnail": "https://books.google.com/books/content?id=MvaNEAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
    "title": "Transcendent Kingdom",
  },
  {
    "authors": [
      "Anna Wiener",
    ],
    "categories": [
      "Biography & Autobiography",
    ],
    "dateFinished": "2020-11-22",
    "description": "One of Vogue''s 22 Books to Read This Winter "A definitive document of a world in transition: I won''t be alone in returning to Uncanny Valley for clarity and consolation for many years to come." —Jia Tolentino, author of Trick Mirror: Reflections on Self-Delusion The prescient, page-turning account of a journey in Silicon Valley: a defining memoir of our digital age In her mid-twenties, at the height of tech industry idealism, Anna Wiener—stuck, broke, and looking for meaning in her work, like any good millennial--left a job in book publishing for the promise of the new digital economy. She moved from New York to San Francisco, where she landed at a big-data startup in the heart of the Silicon Valley bubble: a world of surreal extravagance, dubious success, and fresh-faced entrepreneurs hell-bent on domination, glory, and, of course, progress. Anna arrived amidst a massive cultural shift, as the tech industry rapidly transformed into a locus of wealth and power rivaling Wall Street. But amid the company ski vacations and in-office speakeasies, boyish camaraderie and ride-or-die corporate fealty, a new Silicon Valley began to emerge: one in far over its head, one that enriched itself at the expense of the idyllic future it claimed to be building. Part coming-age-story, part portrait of an already-bygone era, Anna Wiener’s memoir is a rare first-person glimpse into high-flying, reckless startup culture at a time of unchecked ambition, unregulated surveillance, wild fortune, and accelerating political power. With wit, candor, and heart, Anna deftly charts the tech industry’s shift from self-appointed world savior to democracy-endangering liability, alongside a personal narrative of aspiration, ambivalence, and disillusionment. Unsparing and incisive, Uncanny Valley is a cautionary tale, and a revelatory interrogation of a world reckoning with consequences its unwitting designers are only beginning to understand.",
    "isbn": "9780374719760",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=5fqTDwAAQBAJ",
    "pageCount": 288,
    "printType": "BOOK",
    "publishedDate": "2020-01-14",
    "thumbnail": "https://books.google.com/books/content?id=5fqTDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Uncanny Valley",
  },
  {
    "authors": [
      "Kimberly Drew",
      "Jenna Wortham",
    ],
    "categories": [
      "Social Science",
    ],
    "dateFinished": "2021-01-10",
    "description": "NEW YORK TIMES EDITORS’ CHOICE • An archive of collective memory and exuberant testimony A luminous map to navigate an opaque and disorienting present An infinite geography of possible futures What does it mean to be Black and alive right now? Kimberly Drew and Jenna Wortham have brought together this collection of work—images, photos, essays, memes, dialogues, recipes, tweets, poetry, and more—to tell the story of the radical, imaginative, provocative, and gorgeous world that Black creators are bringing forth today. The book presents a succession of startling and beautiful pieces that generate an entrancing rhythm: Readers will go from conversations with activists and academics to memes and Instagram posts, from powerful essays to dazzling paintings and insightful infographics. In answering the question of what it means to be Black and alive, Black Futures opens a prismatic vision of possibility for every reader.",
    "isbn": "9780399181146",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=pUGGDwAAQBAJ",
    "pageCount": 544,
    "printType": "BOOK",
    "publishedDate": "2020-12-01",
    "thumbnail": "https://books.google.com/books/content?id=pUGGDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Black Futures",
  },
  {
    "authors": [
      "Silvia Moreno-Garcia",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2021-09-26",
    "description": "NEW YORK TIMES BESTSELLER • “It’s Lovecraft meets the Brontës in Latin America, and after a slow-burn start Mexican Gothic gets seriously weird.”—The Guardian IN DEVELOPMENT AS A HULU ORIGINAL LIMITED SERIES PRODUCED BY KELLY RIPA AND MARK CONSUELOS • FINALIST FOR THE LOCUS AWARD • NOMINATED FOR THE BRAM STOKER AWARD • NAMED ONE OF THE BEST BOOKS OF THE YEAR BY The New Yorker • Vanity Fair • NPR • The Washington Post • Tordotcom • Marie Claire • Vox • Mashable • Men’s Health • Library Journal • Book Riot • LibraryReads An isolated mansion. A chillingly charismatic aristocrat. And a brave socialite drawn to expose their treacherous secrets. . . . From the author of Gods of Jade and Shadow comes “a terrifying twist on classic gothic horror” (Kirkus Reviews) set in glamorous 1950s Mexico. After receiving a frantic letter from her newly-wed cousin begging for someone to save her from a mysterious doom, Noemí Taboada heads to High Place, a distant house in the Mexican countryside. She’s not sure what she will find—her cousin’s husband, a handsome Englishman, is a stranger, and Noemí knows little about the region. Noemí is also an unlikely rescuer: She’s a glamorous debutante, and her chic gowns and perfect red lipstick are more suited for cocktail parties than amateur sleuthing. But she’s also tough and smart, with an indomitable will, and she is not afraid: Not of her cousin’s new husband, who is both menacing and alluring; not of his father, the ancient patriarch who seems to be fascinated by Noemí; and not even of the house itself, which begins to invade Noemi’s dreams with visions of blood and doom. Her only ally in this inhospitable abode is the family’s youngest son. Shy and gentle, he seems to want to help Noemí, but might also be hiding dark knowledge of his family’s past. For there are many secrets behind the walls of High Place. The family’s once colossal wealth and faded mining empire kept them from prying eyes, but as Noemí digs deeper she unearths stories of violence and madness. And Noemí, mesmerized by the terrifying yet seductive world of High Place, may soon find it impossible to ever leave this enigmatic house behind. “It’s as if a supernatural power compels us to turn the pages of the gripping Mexican Gothic.”—The Washington Post “Mexican Gothic is the perfect summer horror read, and marks Moreno-Garcia with her hypnotic and engaging prose as one of the genre’s most exciting talents.”—Nerdist “A period thriller as rich in suspense as it is in lush ’50s atmosphere.”—Entertainment Weekly",
    "isbn": "9780525620792",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=ksKyDwAAQBAJ",
    "pageCount": 320,
    "printType": "BOOK",
    "publishedDate": "2020-06-30",
    "thumbnail": "https://books.google.com/books/content?id=ksKyDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Mexican Gothic",
  },
  {
    "authors": [
      "bell hooks",
    ],
    "categories": [
      "Social Science",
    ],
    "dateFinished": "2022-01-13",
    "description": "A classic work of feminist scholarship, Ain't I a Woman has become a must-read for all those interested in the nature of black womanhood. Examining the impact of sexism on black women during slavery, the devaluation of black womanhood, black male sexism, racism among feminists, and the black woman's involvement with feminism, hooks attempts to move us beyond racist and sexist assumptions. The result is nothing short of groundbreaking, giving this book a critical place on every feminist scholar's bookshelf.",
    "isbn": "9781317588610",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=f2_fBQAAQBAJ",
    "pageCount": 206,
    "printType": "BOOK",
    "publishedDate": "2014-12-17",
    "thumbnail": "https://books.google.com/books/content?id=f2_fBQAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Ain't I a Woman",
  },
  {
    "authors": [
      "Kali Fajardo-Anstine",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2022-09-07",
    "description": "NATIONAL BESTSELLER • A dazzling epic of betrayal, love, and fate that spans five generations of an Indigenous Chicano family in the American West, from the author of the National Book Award finalist Sabrina & Corina A Phenomenal Book Club Pick • “Sometimes you just step into a book and let it wash over you, like you’re swimming under a big, sparkling night sky.”—Celeste Ng, author of Little Fires Everywhere and Everything I Never Told You ONE OF THE MOST ANTICIPATED BOOKS OF 2022—The Millions, Electric Lit, Lit Hub, Book Riot There is one every generation, a seer who keeps the stories. Luz “Little Light” Lopez, a tea leaf reader and laundress, is left to fend for herself after her older brother, Diego, a snake charmer and factory worker, is run out of town by a violent white mob. As Luz navigates 1930s Denver, she begins to have visions that transport her to her Indigenous homeland in the nearby Lost Territory. Luz recollects her ancestors’ origins, how her family flourished, and how they were threatened. She bears witness to the sinister forces that have devastated her people and their homelands for generations. In the end, it is up to Luz to save her family stories from disappearing into oblivion. Written in Kali Fajardo-Anstine’s singular voice, the wildly entertaining and complex lives of the Lopez family fill the pages of this multigenerational western saga. Woman of Light is a transfixing novel about survival, family secrets, and love—filled with an unforgettable cast of characters, all of whom are just as special, memorable, and complicated as our beloved heroine, Luz.",
    "isbn": "9780525511342",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=5LhBEAAAQBAJ",
    "pageCount": 336,
    "printType": "BOOK",
    "publishedDate": "2022-06-07",
    "thumbnail": "https://books.google.com/books/content?id=5LhBEAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Woman of Light",
  },
  {
    "authors": [
      "Kim Fu",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2022-09-11",
    "dateStarted": "2022-09-10",
    "description": ""The strange and wonderful define Kim Fu’s story collection, where the line between fantasy and reality fades in and out, elusive and beckoning." —The New York Times Book Review A LitHub, ALTA, and PureWow Best Book of the Month A BuzzFeed and WIRED Pick for a Book You Need to Read This Winter In the twelve unforgettable tales of Lesser Known Monsters of the 21st Century, the strange is made familiar and the familiar strange, such that a girl growing wings on her legs feels like an ordinary rite of passage, while a bug-infested house becomes an impossible, Kafkaesque nightmare. Each story builds a new world all its own: a group of children steal a haunted doll; a runaway bride encounters a sea monster; a vendor sells toy boxes that seemingly control the passage of time; an insomniac is seduced by the Sandman. These visions of modern life wrestle with themes of death and technological consequence, guilt and sexuality, and unmask the contradictions that exist within all of us. Mesmerizing, electric, and wholly original, Kim Fu’s Lesser Known Monsters of the 21st Century blurs the boundaries of the real and fantastic, offering intricate and surprising insights into human nature.",
    "isbn": "9781953534064",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=4D1UEAAAQBAJ",
    "notes": "",
    "pageCount": 232,
    "printType": "BOOK",
    "publishedDate": "2022-02-01",
    "thumbnail": "https://books.google.com/books/content?id=4D1UEAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Lesser Known Monsters of the 21st Century",
  },
  {
    "authors": [
      "Octavia E. Butler",
    ],
    "categories": [
      "Fiction",
    ],
    "dateAdded": "2022-10-02",
    "description": ""A master storyteller, Butler casts an unflinching eye on racism, sexism, poverty, and ignorance and lets the reader see the terror and beauty of human nature." - The Washington Post This is the story of an apparently young, amnesiac girl whose alarmingly unhuman needs and abilities lead her to a startling conclusion: She is in fact a genetically modified, 53-year-old vampire. Forced to discover what she can about her stolen former life, she must at the same time learn who wanted-and still wants-to destroy her and those she cares for and how she can save herself.",
    "isbn": "9780446696166",
    "language": "en",
    "link": "https://books.google.com/books/about/Fledgling.html?hl=&id=LIH6wAEACAAJ",
    "pageCount": 320,
    "printType": "BOOK",
    "publishedDate": "2007-01-02",
    "status": "want to read",
    "thumbnail": "https://books.google.com/books/content?id=LIH6wAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
    "title": "Fledgling",
  },
  {
    "authors": [
      "Ling Ma",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2022-11-25",
    "dateStarted": "2022-11-24",
    "description": "A new creation by the author of Severance, Bliss Montage crashes through our carefully built mirages. What happens when fantasy tears the screen of the everyday to wake us up? Could that waking be our end? In Bliss Montage, Ling Ma brings us eight wildly different tales of people making their way through the madness and reality of our collective delusions: love and loneliness, connection and possession, friendship, motherhood, the idea of home. A woman lives in a house with all her ex-boyfriends. A toxic friendship grows up around a drug that makes you invisible. An ancient ritual might heal you of anything—if you bury yourself alive. These and other scenarios investigate the ways that the outlandish and the ordinary are shockingly, deceptively, heartbreakingly alike.",
    "isbn": "9780374717124",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=8ABdEAAAQBAJ",
    "pageCount": 145,
    "printType": "BOOK",
    "publishedDate": "2022-09-13",
    "rating": "⭐️⭐️⭐️⭐️",
    "status": "finished",
    "tags": [
      "recommend",
    ],
    "thumbnail": "https://books.google.com/books/content?id=8ABdEAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Bliss Montage",
  },
  {
    "authors": [
      "Da'Shaun L. Harrison",
    ],
    "categories": [
      "Social Science",
    ],
    "dateFinished": "2022-11-30",
    "dateStarted": "2022-11-30",
    "description": "**The 2022 Lammy Award Winner in Transgender Nonfiction** Exploring the intersections of Blackness, gender, fatness, health, and the violence of policing. To live in a body both fat and Black is to exist at the margins of a society that creates the conditions for anti-fatness as anti-Blackness. Hyper-policed by state and society, passed over for housing and jobs, and derided and misdiagnosed by medical professionals, fat Black people in the United States are subject to sociopolitically sanctioned discrimination, abuse, condescension, and trauma. Da’Shaun Harrison--a fat, Black, disabled, and nonbinary trans writer--offers an incisive, fresh, and precise exploration of anti-fatness as anti-Blackness, foregrounding the state-sanctioned murders of fat Black men and trans and nonbinary masculine people in historical analysis. Policing, disenfranchisement, and invisibilizing of fat Black men and trans and nonbinary masculine people are pervasive, insidious ways that anti-fat anti-Blackness shows up in everyday life. Fat people can be legally fired in 49 states for being fat; they’re more likely to be houseless. Fat people die at higher rates from misdiagnosis or nontreatment; fat women are more likely to be sexually assaulted. And at the intersections of fatness, Blackness, disability, and gender, these abuses are exacerbated. Taking on desirability politics, the limitations of gender, the connection between anti-fatness and carcerality, and the incongruity of “health” and “healthiness” for the Black fat, Harrison viscerally and vividly illustrates the myriad harms of anti-fat anti-Blackness. They offer strategies for dismantling denial, unlearning the cultural programming that tells us “fat is bad,” and destroying the world as we know it, so the Black fat can inhabit a place not built on their subjugation.",
    "isbn": "9781623175979",
    "language": "en",
    "link": "https://books.google.com/books/about/Belly_of_the_Beast.html?hl=&id=nvQ3EAAAQBAJ",
    "pageCount": 148,
    "printType": "BOOK",
    "publishedDate": "2021-08-10",
    "rating": "⭐️⭐️⭐️⭐️⭐️",
    "status": "finished",
    "tags": [
      "recommend",
    ],
    "thumbnail": "https://books.google.com/books/content?id=nvQ3EAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Belly of the Beast",
  },
  {
    "authors": [
      "Rob Hart",
    ],
    "categories": [
      "Fiction",
    ],
    "dateAdded": "2022-11-25",
    "dateFinished": "2022-12-03",
    "dateStarted": "2022-11-30",
    "description": "“Time travel, murder, corruption, restless baby dinosaurs, and a snarky robot named Ruby collide in this excellent, noir-inflected, humor-infused, science-fiction thriller.”—The Boston Globe An impossible crime. A detective on the edge of madness. The future of time travel at stake. From the author of The Warehouse . . . ONE OF THE BEST BOOKS OF THE YEAR: NPR January Cole’s job just got a whole lot harder. Not that running security at the Paradox was ever really easy. Nothing’s simple at a hotel where the ultra-wealthy tourists arrive costumed for a dozen different time periods, all eagerly waiting to catch their “flights” to the past. Or where proximity to the timeport makes the clocks run backward on occasion—and, rumor has it, allows ghosts to stroll the halls. None of that compares to the corpse in room 526. The one that seems to be both there and not there. The one that somehow only January can see. On top of that, some very important new guests have just checked in. Because the U.S. government is about to privatize time-travel technology—and the world’s most powerful people are on hand to stake their claims. January is sure the timing isn’t a coincidence. Neither are those “accidents” that start stalking their bidders. There’s a reason January can glimpse what others can’t. A reason why she’s the only one who can catch a killer who’s operating invisibly and in plain sight, all at once. But her ability is also destroying her grip on reality—and as her past, present, and future collide, she finds herself confronting not just the hotel’s dark secrets but her own. At once a dazzlingly time-twisting murder mystery and a story about grief, memory, and what it means to—literally—come face-to-face with our ghosts, The Paradox Hotel is another unforgettable speculative thrill ride from acclaimed author Rob Hart.",
    "isbn": "9781984820648",
    "language": "en",
    "link": "https://books.google.com/books/about/The_Paradox_Hotel.html?hl=&id=8bNaEAAAQBAJ",
    "pageCount": 337,
    "printType": "BOOK",
    "publishedDate": "2022-02-22",
    "rating": "⭐️⭐️⭐️",
    "status": "finished",
    "thumbnail": "https://books.google.com/books/content?id=8bNaEAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "The Paradox Hotel",
  },
  {
    "authors": [
      "Jennifer Egan",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2022-12-06",
    "dateStarted": "2022-12-03",
    "description": "Named a Most Anticipated Book of the Year by Time, Entertainment Weekly, Vogue, Good Housekeeping, Oprah Daily, Glamour, USA TODAY, Parade, Bustle, San Francisco Chronicle, The Seattle Times, The Boston Globe, Tampa Bay Times, BuzzFeed, Vulture, and many more! From one of the most celebrated writers of our time, a literary figure with cult status, a "sibling novel" to her Pulitzer Prize- and ?NBCC Award-winning A Visit from the Goon Squad--an electrifying, deeply moving novel about the quest for authenticity and meaning in a world where memories and identities are no longer private. The Candy House opens with the staggeringly brilliant Bix Bouton, whose company, Mandala, is so successful that he is "one of those tech demi-gods with whom we're all on a first name basis." Bix is 40, with four kids, restless, desperate for a new idea, when he stumbles into a conversation group, mostly Columbia professors, one of whom is experimenting with downloading or "externalizing" memory. It's 2010. Within a decade, Bix's new technology, "Own Your Unconscious"--that allows you access to every memory you've ever had, and to share every memory in exchange for access to the memories of others--has seduced multitudes. But not everyone. In spellbinding interlocking narratives, Egan spins out the consequences of Own Your Unconscious through the lives of multiple characters whose paths intersect over several decades. Intellectually dazzling, The Candy House is also extraordinarily moving, a testament to the tenacity and transcendence of human longing for real connection, love, family, privacy and redemption. In the world of Egan's spectacular imagination, there are "counters" who track and exploit desires and there are "eluders," those who understand the price of taking a bite of the Candy House. Egan introduces these characters in an astonishing array of narrative styles--from omniscient to first person plural to a duet of voices, an epistolary chapter and a chapter of tweets. If Goon Squad was organized like a concept album, The Candy House incorporates Electronic Dance Music's more disjunctive approach. The parts are titled: Build, Break, Drop. With an emphasis on gaming, portals, and alternate worlds, its structure also suggests the experience of moving among dimensions in a role-playing game. The Candy House is a bold, brilliant imagining of a world that is moments away. Egan takes to stunning new heights her "deeply intuitive forays into the darker aspects of our technology-driven, image-saturated culture" (Vogue). The Candy House delivers an absolutely extraordinary combination of fierce, exhilarating intelligence and heart.",
    "isbn": "9781476716763",
    "language": "en",
    "link": "https://books.google.com/books/about/The_Candy_House.html?hl=&id=WB9oEAAAQBAJ",
    "pageCount": 352,
    "printType": "BOOK",
    "publishedDate": "2022-04-05",
    "rating": "⭐️⭐️⭐️⭐️",
    "status": "finished",
    "tags": [
      "recommend",
    ],
    "thumbnail": "https://books.google.com/books/content?id=WB9oEAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "The Candy House",
  },
]
`;
